// bitcore/networks.js
var Put = require('bufferput');
var buffertools = require('buffertools');
var hex = function(hex) {return new Buffer(hex, 'hex');};

exports.livenet = {
  name: 'livenet',
  magic: hex('fbc0b6db'),
  //magic: hex('f9beb4d9'),
  //magic: hex('D9B4BEF9'),
  //addressVersion: 0x17,
  addressVersion:125,
  //privKeyVersion: 151,
  privKeyVersion:128,
  P2SHVersion: 5,
  hkeyPublicVersion: 0x0488b21e,
  hkeyPrivateVersion: 0x0488ade4,
  genesisBlock: {
     //hash: hex('e2bf047e7e5a191aa4ef34d314979dc9986e0f19251edaba5940fd1fe365a712'),
    //merkle_root: hex('d9ced4ed1130f7b7faad9be25323ffafa33232a17c3edf6cfd97bee6bafbdd97'),
    hash: hex('cff061b4711220a274d17c658513da569a9d60bd04e9b5fa85b151cdc63f6b3a'),
    merkle_root: hex('971c136cbf60c3592ed62fb89f8a1ba7fa994f77182b0cc9258f4d893b776ff6'),
    height: 0,
    //nonce: 4242852485,
    nonce: 2084524493,
    version: 1,
    prev_hash: buffertools.fill(new Buffer(32), 0),
    //timestamp: 1398689365,
    timestamp: 1317972665,
    bits: hex('1e0ffff0'),
  //bits: 504365040,
  },
  dnsSeeds: [
 // 'dnsseed.litecointools.com',
 // 'dnsseed.litecoinpool.org',
 // 'dnsseed.ltc.xurious.com',
 // 'dnsseed.koin-project.com',
 // 'dnsseed.weminemnc.com',
    'dnsseed1.suwoncoin.org',
    'dnsseed2.suwoncoin.org',
    'dnsseed3.suwoncoin.org', 
],
   defaultClientPort: 8608
 //defaultClientPort: 9777
};

exports.testnet = {
  name: 'testnet',
  magic: hex('0b110907'),
  addressVersion: 0x6f,
  privKeyVersion: 239,
  P2SHVersion: 196,
  hkeyPublicVersion: 0x043587cf,
  hkeyPrivateVersion: 0x04358394,
  genesisBlock: {
    hash: hex('43497FD7F826957108F4A30FD9CEC3AEBA79972084E90EAD01EA330900000000'),
    merkle_root: hex('3BA3EDFD7A7B12B27AC72C3E67768F617FC81BC3888A51323A9FB8AA4B1E5E4A'),
    height: 0,
    nonce: 414098458,
    version: 1,
    prev_hash: buffertools.fill(new Buffer(32), 0),
    timestamp: 1296688602,
    bits: 486604799,
  },
  dnsSeeds: [
    'testnet-seed.bitcoin.petertodd.org',
    'testnet-seed.bluematt.me'
  ],
  defaultClientPort: 18608
};
